localization/pose_estimator_arbiter/test/pose_estimator_arbiter.test.py

```diff
@@ -0,0 +1,207 @@
+            lambda msg: self.ndt_relayed.append(msg.header),
+            qos_profile,
+        )
+        self.test_node.create_subscription(
+            Image,
+            "/sensing/camera/traffic_light/image_raw/yabloc_relay",
+            lambda msg: self.yabloc_relayed.append(msg.header),
+            qos_profile,
+        )
+
+    def test_node_link(self):
+        # The arbiter waits for the service to start, so here it instantiates a meaningless service server.
+        self.test_node.create_service(
+            SetBool,
+            "/localization/pose_estimator/yabloc/pf/yabloc_suspend_srv",
+            self.yabloc_suspend_service_callback,
+        )
+
+        # Define subscription buffer
+        self.ndt_relayed = []
+        self.yabloc_relayed = []
+
+        # Create publishers and subscribers
+        self.create_publishers_and_subscribers()
+
+        # Wait 0.5 second for node to be ready
+        self.spin_for(0.5)
+
+        # Publish dummy input topics
+        for _ in range(10):
+            self.publish_input_topics()
+            rclpy.spin_once(self.test_node, timeout_sec=0.1)
+
+        # Wait 1.0 second for all topics to be subscribed
+        self.spin_for(1.0)
+
+        # Confirm both topics are relayed
+        print("ndt", len(self.ndt_relayed))
+        print("yabloc", len(self.yabloc_relayed))
+        self.assertGreater(len(self.ndt_relayed), 5)
+        self.assertGreater(len(self.yabloc_relayed), 5)
+
+        # Change UNINITIALIZED -> INITIALIZED
+        pub_init_state = self.test_node.create_publisher(
+            LocalizationInitializationState,
+            "/localization/initialization_state",
+            QoSProfile(depth=10, durability=DurabilityPolicy.TRANSIENT_LOCAL),
+        )
+        init_state = LocalizationInitializationState()
+        init_state._state = LocalizationInitializationState.INITIALIZED
+        pub_init_state.publish(init_state)
+
+        # Publish dummy estimated pose
+        pub_pose_stamped = self.test_node.create_publisher(
+            PoseWithCovarianceStamped,
+            "/localization/pose_with_covariance",
+            10,
+        )
+        pub_pose_stamped.publish(PoseWithCovarianceStamped())
+
+        # Because pointcloud_map is not broadcasted, the arbiter state must be following:
+        #  - ndt: disabled
+        #  - yabloc: enabled
+        #  - eagleye: disabled
+        #  - artag: disabled
+
+        # Wait 1.0 second for the node state to change
+        self.spin_for(1.0)
+
+        # Clear buffer
+        self.ndt_relayed = []
+        self.yabloc_relayed = []
+
+        # Publish dummy input topics
+        for _ in range(10):
+            self.publish_input_topics()
+            rclpy.spin_once(self.test_node, timeout_sec=0.1)
+
+        # Wait 0.5 second for all topics to be subscribed
+        self.spin_for(0.5)
+
+        # Confirm that yabloc topic is relayed and ndt toic is not relayed.
```

Comment:
There seems to be a typo.

Reply:
Fixed https://github.com/autowarefoundation/autoware.universe/pull/5846/commits/aae8e8e1c94510c31f2d7dc76d1bb3b5a50c54a2